src/engine/strat_engine/lineardev.rs

```diff
@@ -0,0 +1,66 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use devicemapper::{DM, DevId, DeviceInfo, DmFlags};
+
+use engine::{EngineError, EngineResult, ErrorEnum};
+use super::blockdev::BlockDev;
+use types::Sectors;
+
+pub struct LinearDev {
+    name: String,
```

Comment:
The traits here: https://github.com/stratis-storage/stratisd/blob/master/src/engine/engine.rs are almost adequately commented.

The tests here: https://github.com/stratis-storage/stratisd/blob/master/src/engine/strat_engine/metadata.rs#L533 state what they're trying to do reasonably well.